controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -43,6 +43,54 @@ static std::string check_forbidden_chars(const std::string &str)
     return forbidden_chars_found;
 }
 
+/*
+* @brief Convert a binary string to hex notation 
+* 
+* @param[in] binary_string String in binary notation
+* @return None
+*/
+static void binary_to_hex(std::string &binary_string)
+{
+    std::string output("0x");
+    uint8_t idx = 2;
+
+    while (idx < binary_string.length()) {
+        // Read 8 bits
+        auto binary_byte = binary_string.substr(idx, 8);
+
+        // Convert binary byte to integer
+        auto integer_byte = std::stoi(binary_byte, nullptr, 2);
+
+        // Convert integer to hex string with 2 characters and add it to output
+        output += string_utils::int_to_hex_string(integer_byte, 2);
+        idx += 8;
+    }
+
+    binary_string = output;
+}
+
+/*
+* @brief convert a decimal string to hex notation 
+* 
+* @param[in] decimal_string String in decimal notation
+* @return None
+*/
+static void decimal_to_hex(std::string &decimal_string)
+{
+    std::string output("0x");
+    decimal_string.erase(0, 2);
+
+    int i_dec        = std::stoi(decimal_string);
+    int rem          = i_dec % 255;
+    int num_of_bytes = i_dec / 255;
+    if (rem != 0) {
+        num_of_bytes++;
+    }
+
+    output += string_utils::int_to_hex_string(i_dec, num_of_bytes);
```

Comment:
The decimal case is tricky... For hex and binary, you can know the length from the 0-padded number of characters. For decimal, this doesn't really work though. Therefore, I assume that the decimal case is only used in practice for 1-byte values. Is that possible?

Reply:
I have seen at least one case of a decimal number that requires more than one byte for storage, e.g., 0d0600  in test 4.8.5 in Test Plan.

Reply:
> I have seen at least one case of a decimal number that requires more than one byte for storage, e.g., 0d0600 in test 4.8.5 in Test Plan.

Darn. So it means that this code has to stay. But it's insane... for 0d0600 it happens to work out OK with 2 bytes, but what would you do with 0d0200? It would end up as one byte... Whoever thought up this CAPI spec wasn't very awake...